### ISSUE TITLE
Marker method setIcon make map and all markers not draggable when I'm using it with "contextmenu" event

### ISSUE THREAD

**Issue by samgold172** [Leaflet/Leaflet#5647]
## How to reproduce

- Leaflet version I'm using: 1.1.0
- Browser (with version) I'm using: Chrome 59.0.3071.115
- OS/Platform (with version) I'm using: Ubuntu 16.04
- Add event listener to marker "contextmenu"
- Call marker method "setIcon" on"contextmenu" event this.setIcon(L.icon({iconUrl: 'myicon.png'}))

## What behaviour I'm expecting and which behaviour I'm seeing
I'm expecting just to change marker icon, but map and all markers become not draggable.

## Minimal example reproducing the issue
Sample 
https://jsfiddle.net/vitaliy_korchak/h299wLtn/2/

**Comment by jessenic** [comment]
Probably related to #5638

**Comment by codeofsumit** [comment]
I think this broke my plugin leaflet.pm as well. 
When removing a marker, there is no drag anymore except some custom drag behavior that I've implemented myself 🤔
I'm still investigating. 
Works fine in 1.0.3

**Comment by jjimenezshaw** [comment]
Yes, it is the same problem as #5638 (that is now closed), and is solved in 1.2.0 as well.
I forked the jsfiddle, upgrading to 1.2.0, and changed the icon (to the shadow one) and added a tooltip to see that something happens.
https://jsfiddle.net/vqhssxvc/
